<?xml version="1.0" ?>

<!--

Binary sonar emulation

-->


<launch>
	
  <node pkg="ros_pure" type="ros_pure_bridge.py" name="ros_pure_bridge" output="screen"> 
  </node>


  <node pkg="tf" type="static_transform_publisher" name="laser2robot" args="0 0 0 0 0 0 /base_link /kompai_laser 100"/>
  <node pkg="tf" type="static_transform_publisher" name="odom2map" args="0 0 0 0 0 0 map odom_combined 100"/>
    
  <node pkg="rfid_node" type="rfid_detect_eu.py" name="tag_detect_node" output="screen">
        <rosparam param="txpower">2000</rosparam>	            
    </node>   
  
  <group ns="tag_resources_0">	  
	<node pkg="rfid_node" type="binary_tag_sonar.py" name="tag_sonar_node" output="screen">
        <rosparam param="tagNAME">"390000010000000000000007"</rosparam>	    
        <rosparam param="tsample">1</rosparam>	    
        <remap from="/tag_resources_0/lastTag" to="/lastTag"/>
    </node>   

	<node pkg="tf" type="static_transform_publisher" name="tag2robot" args="0 0 0 0 0 0  base_link sonar_390000010000000000000007  100" />
<!--   Run the costmap node -->
	<node name="costmap_node" pkg="costmap_2d" type="costmap_2d_node" output="screen" >
	  <rosparam file="$(find rfid_node)/launch/costmap_params_binary.yaml" command="load" ns="costmap" />
	</node> 
  </group>  

  <node name="costmap_node" pkg="costmap_2d" type="costmap_2d_node" output="screen" >
	  <rosparam file="$(find rfid_node)/launch/costmap_params_laser.yaml" command="load" ns="costmap" />
  </node>

</launch>
